pkg/scheduler/framework/types.go

```diff
@@ -40,6 +40,52 @@ import (
 
 var generation int64
 
+// ActionType is an integer to represent one type of resource change.
+// Different ActionTypes can be bit-wised to compose new semantics.
+type ActionType int64
+
+// Constants for ActionTypes.
+const (
+	Add    ActionType = 1 << iota // 1
+	Delete                        // 10
+	// UpdateNodeXYZ is only applicable for Node events.
+	UpdateNodeAllocatable // 100
+	UpdateNodeLabel       // 1000
+	UpdateNodeTaint       // 10000
+	UpdateNodeCondition   // 100000
+
+	All ActionType = 1<<iota - 1 // 111111
+
+	// Use the general Update type if you don't either know or care the specific sub-Update type to use.
+	Update = UpdateNodeAllocatable | UpdateNodeLabel | UpdateNodeTaint | UpdateNodeCondition
+)
+
+// GVK is short for group/version/kind, which can uniquely represent a particular API resource.
+type GVK string
+
+// Constants for GVKs.
+const (
+	Pod                   GVK = "Pod"
+	Node                  GVK = "Node"
+	PersistentVolume      GVK = "PersistentVolume"
+	PersistentVolumeClaim GVK = "PersistentVolumeClaim"
+	Service               GVK = "Service"
+	StorageClass          GVK = "storage.k8s.io/StorageClass"
+	CSINode               GVK = "storage.k8s.io/CSINode"
+	WildCard              GVK = "*"
+)
+
+// WildCardEvent semantically matches all resources on all actions.
```

Comment:
all ones we registered event handlers for? or literally everything?

Reply:
It can be used in both event handlers side and plugin side, a bit rare in plugin side though as plugin can choose to not implement `EventsToRegister()` to achieve the same goal.

- In the event handlers side, it implies an enforced moving. For now, it's used to signal flushing unschedulable Pods upon timeout.
- In the plugin side, it literally means to register all cluster events.